src/main/java/uk/gov/justice/digital/delius/service/Result.java

```diff
@@ -0,0 +1,38 @@
+package uk.gov.justice.digital.delius.service;
+
+import java.util.NoSuchElementException;
+import java.util.function.Function;
+
+public class Result<T, X extends Throwable> {
+    private T value;
+    private X error;
+
+    public static <T, X extends Throwable> Result<T, X> of(T value) {
+        return new Result<>(value);
+    }
+    public static <T, X extends Throwable> Result<T, X> ofError(X error) {
+        return new Result<>(error);
+    }
+    private Result(T value) {
+        this.value = value;
+    }
+    private Result(X error) {
+        this.error = error;
+    }
+
+    public T onError(Function<X, ? extends RuntimeException> exceptionSupplier) {
+        if (value != null) {
+            return value;
+        } else {
```

Comment:
Meh you don't need the else....

Reply:
Interestingly JDK do use else - from Optional
`    public <X extends Throwable> T orElseThrow(Supplier<? extends X> exceptionSupplier) throws X {
        if (value != null) {
            return value;
        } else {
            throw exceptionSupplier.get();
        }
    }`

I was taught not to - oh well ...